### PR TITLE
Feature/authorization

### PR DESCRIPTION
**Pull request by dusanstanis**
Added authorization with roles and permissions in our system.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -146,6 +149,12 @@
     "source-map-loader": "0.2.4",
     "supertest": "4.0.2",
     "swagger-jsdoc": "3.2.7",
+    "swagger-ui-express": "4.0.2",
+    "ts-node": "^10.2.1",
+    "typescript": "^4.4.2"
+  },
+  "eslintConfig": {
+    "extends": "eslint:recommended",
```
Comment (rekpero) [inline package.json:157]: why is this changed? please remove the `eslintConfig`

**Review by rekpero — CHANGES_REQUESTED** [review]
@dusanstanis PR lgtm!! please check the package json and revert the eslintConfig

File: src/routes/AuthRouter.ts [hunk 2]
```diff
@@ -82,6 +81,19 @@ router.get(
     );
   }
 );
+router.post("/fakeAuth", async (req, res) => {
```
Comment (rekpero) [inline src/routes/AuthRouter.ts:84]: why is this needed?

File: src/routes/DeploymentsRouter.ts [hunk 3]
```diff
@@ -1,9 +1,14 @@
 import { Router } from "express";
 import { DeploymentComponent } from "../components";
+import { grantAccess } from "../config/middleware/validate";
 
 const router: Router = Router();
 
-router.post("/", DeploymentComponent.deployFromRequest);
+router.post(
+  "/",
+  grantAccess("deploymentAllowance"),
```
Comment (rekpero) [inline src/routes/DeploymentsRouter.ts:9]: create these `deploymentAllowance` access types in an enum type and pass the enum here. it's much better than having string here

File: src/routes/RoleRouter.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+import { grantAccess } from "../config/middleware/validate";
+
+const router: Router = Router();
+
+router.post("/insert", RoleComponent.insert);
+
+router.post(
+  "/addPermission",
+  grantAccess("permissionManagementAllowance"),
```
Comment (rekpero) [inline src/routes/RoleRouter.ts:12]: same here, let's put these on a enum and use the enum to grant access to the API calls

File: src/routes/SubscriptionRouter.ts [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
+router.post(
+  "/cancelSubscription",
+  grantAccess("subscriptionAllowance"),
+  SubscriptionComponent.cancelSubscription
+);
 
-router.post("/subscribe", SubscriptionComponent.subscribe);
+router.post(
+  "/subscribe",
+  grantAccess("subscriptionAllowance"),
```
Comment (rekpero) [inline src/routes/SubscriptionRouter.ts:15]: same here

File: package.json (hunk 1 above)
Comment (rekpero) [reply package.json:166]: @nikoleta-vukajlovic @dusanstanis why is this still left

**Review by rekpero — APPROVED** [review]

File: src/components/Auth/service.ts [hunk 6]
```diff
@@ -8,6 +8,10 @@ import { IAuthService } from "./interface";
 import { Request } from "express";
 import JWTTokenService from "../Session/service";
 import * as Email from "../Email/index";
+import { IRole, RoleGroupsEnum } from "../Role/interfaces";
+import RoleService from "../Role/service";
```
Comment (akinovak) [inline src/components/Auth/service.ts:12]: Should we mix different services in service.ts or add component with index.ts?

**Review by akinovak — APPROVED** [review]

**Pull request merged**